### ISSUE TITLE
Argument alignment

### ISSUE THREAD

**Issue by mjago** [asterite/rufo#91]
I found this interesting snippet:
```ruby
wx yz(:one,
  :two,
  :three)
```
...formats to:
```ruby
wx yz(:one,
     :two,
     :three)
```
...formats to:
```ruby
wx yz(:one,
      :two,
      :three)
```

**Comment by asterite** [comment]
Thanks!

I have a few other alignment bugs.. it will probably take me a lot of time to fix them all. I'll go slowly :-) (and PRs are welcome, though I'm starting to realize there are way too many valid formatting options that Rubyists like...)

**Comment by mjago** [comment]
👍 I wondered if this pointed to a simple out by 1 error so thought I should mention it.

**Comment by asterite** [comment]
I also wonder what the correct alignment should be. Probably 2, 3 or 5 spaces of indentation are all acceptable.

**Comment by mjago** [comment]
I agree and have no preference. Are there cases where valid consecutive formats are feasible. I assumed that any 'reformatting' of previously formatted code points to a potential error (akin to a race condition). I think that would be possible if 'look-ahead' formatting were involved?

**Comment by asterite** [comment]
What I mean is that this code:

```ruby
wx yz(:one,
  :two,
  :three)
```

should be left untouched, because it's correctly indented: 2 spaces.

This code:

```ruby
wx yz(:one,
      :two,
      :three)
```

is also correctly indented (aligned to the first parameter of `yz`), so it should be left untouched too.

I'm thinking about:

```ruby
wx yz(:one,
   :two,
   :three)
```

Even though it's aligned to the first parameter of `wx` it's probably not correct because there's a nested call...

In case when the formatter thinks something is not correct, it will change it using one rule. For example in the above case it should change it to the second snippet (align with the first parameter of `yz`)... that is, unless we introduce settings to decide which indentation to prefer in case of an "incorrect" formatting.

I want to fix and do all of that but I just recently realized this "many formats are valid and we need to track positions of parameters" issues. This is already done in some case, like what's explained in the README, but many more are missing.

Right now the code to decide that is starting to become a bit hard to understand, so I'll probably refactor first what there's now.

Anyway, slowly but steadily... :-)

**Comment by mjago** [comment]
Yes I see what you mean. I haven't even looked at the indentation code since it looked a bit complicated 😉 . Perhaps the indentation rules can be extract out into a state-machine (you could at least then draw the decision tree then with graphviz or similar which I find helpful).

**Comment by shioyama** [comment]
I notice something similar:

Before

```ruby
validates :foo, numericality: {
  greater_than_or_equal_to: 0,
  less_than_or_equal_to: 999_999
}
```

After first run through `rufo`:

```ruby
validates :foo, numericality: {
        greater_than_or_equal_to: 0,
        less_than_or_equal_to: 999_999
      }
```

After second run through `rufo`:

```ruby
validates :foo, numericality: {
                  greater_than_or_equal_to: 0,
                  less_than_or_equal_to: 999_999
                }
```

Not really sure why this should happen here.

**Comment by neerajdotname** [comment]
I concur that there are way too many formatting rules in ruby. Perhaps we can learn a few things from prettier and make it more opinionated.
